plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelquery/completion/TargetCompletionsGenerator.kt

```diff
@@ -0,0 +1,120 @@
+package org.jetbrains.bazel.languages.bazelquery.completion
+
+import com.intellij.openapi.project.Project
+import org.jetbrains.bazel.languages.starlark.repomapping.toShortString
+import org.jetbrains.bazel.target.targetUtils
+import java.nio.file.Path
+import kotlin.text.removePrefix
+import kotlin.text.startsWith
+
+class TargetCompletionsGenerator(private val project: Project) {
+  private val separator = "/"
+  private val startTargetSign = "//"
+  private val allTargets =
+    project.targetUtils
+      .allTargets()
+      .map { it.toShortString(project) }
+      .toMutableSet()
+
+  fun getTargetsList(prefix: String, directory: Path? = null): List<String> {
+    // Note: project root dependent format - starts with "//",
+    // directory dependent format - starts with a letter or ":"
+    val suggestions = mutableListOf<String>()
+    val projectPath = Path.of(project.basePath ?: return suggestions)
+    if (directory != null && !directory.startsWith(projectPath)) return suggestions
```

Comment:
so you don't support external dependencies query?
e.g., you can do this: `bazel query "deps(@maven//:com_google_guava_guava, 2)"`